src/core/jbig2.js

```diff
@@ -606,6 +606,136 @@ var Jbig2Image = (function Jbig2ImageClosure() {
     }
     return bitmap;
   }
+  
+  function decodePatternDictionary(mmr, patternWidth, patternHeight,
+                                   maxPatternIndex, template, decodingContext) {
+    var at = [];
+    at.push({
+      x: -patternWidth,
+      y: 0
+    });
+    if (template === 0) {
+      at.push({
+        x: -3,
+        y: -1
+      });
+      at.push({
+        x: 2,
+        y: -2
+      });
+      at.push({
+        x: -2,
+        y: -2
+      });
+    }
+    var collectiveWidth = (maxPatternIndex + 1)*patternWidth;
+    var collectiveBitmap = decodeBitmap(mmr, collectiveWidth, patternHeight,
+                                        template, false, null, at,
+                                        decodingContext);
+    // Divide collective bitmap into patterns
```

Comment:
Nit: please remember to end sentences with a period (applies elsewhere in the patch as well).